### PR TITLE
Update db

### PR DESCRIPTION
**Pull request by Rahaf-96**
- updatd database by adding users table with entries(username, email, password, status and type).
- add one user to the table (admin) in dummy data.
- change files naming to make all their names in camel case.

### CODE REVIEW

File: server/database/config/dbBuild.sql [hunk 1]
```diff
@@ -63,4 +63,15 @@ CREATE TABLE std_projects
...
+(
+  id SERIAL PRIMARY KEY,
+  username VARCHAR(50) NOT NULL,
+  email VARCHAR(255) NOT NULL,
+  password TEXT NOT NULL,
+  status VARCHAR(50),
+  type VARCHAR(50)
+  );
+
+
```
Comment (ghassanmas) [inline server/database/config/dbBuild.sql:76]: I have notes regarding the table: 

Using `UNIQUE` constraint for the column/s which you won't allow having a duplicate value in e.g. do you expect two users to have the same email address... 
Also, this would automatically add an extra layer of protection;  the `pg` module would throw an error if you are trying to add a new user with an email address that already exists in the table.  

Finally, consider applying what I have said above to the `username` column.

Comment (Rahaf-96) [reply server/database/config/dbBuild.sql:76]: @ghassanmas Actually I didn't make it UNIQUE because I didn't want to write each time we test something on the sign up process a different email and username( just to save time :sweat_smile: ), so I wanted to change it after we finish the sign up component. But I will create an issue to remind me of that .

Comment (ghassanmas) [reply server/database/config/dbBuild.sql:76]: @Rahaf-96 sounds reasonable, though I would recommend for each time you run the server/app locally in **development environment**. To rebuild the database which should clean all rows. This for sure assuming _you have two databases at least one for production and one for development/testing_.

Comment (Rahaf-96) [reply server/database/config/dbBuild.sql:76]: @ghassanmas  It's a good idea, I will try it. 
Also I was thinking that in the sign up process we will write a query to check email before storing it in database, like if the email exists send a message to the user that the email already exists and it won't insert a new user in the database. 
So this way we won't have two users with the same email address. 
In this situation do we have to make it unique in sql build?

Comment (ghassanmas) [reply server/database/config/dbBuild.sql:76]: @Rahaf-96 first of all, **it's true and important**  that you gotta have a mechanism to tell the user that their email address already exists if they typed an email that already exists in the signup form. 
Now the question of how to do that is more or less an **opinionated question**. that is subjective to the developer's opinion and application architecture..etc. You could check for it as soon as the user finishes typing the email input field OR when the user hits the submit button... 

Lastly, you still gotta add the `UNIQUE` constraint for different reasons. like security or future app development and integration... for example, if in the future a new developer wants to update your code and they didn't check for email or if the database is connected to a new different server/client. "Would you blindly trust all these scenarios?" 

TLDR; _client or server-side validation shouldn't replace DB constraints_

Comment (Rahaf-96) [reply server/database/config/dbBuild.sql:76]: @ghassanmas Okaay I understand now! Thanks for the explanation 🙏
